app/assets/javascripts/discourse/app/components/composer-body.js

```diff
@@ -62,6 +62,16 @@ export default Component.extend(KeyEnterEscape, {
     this.appEvents.trigger("composer:resized");
   },
 
+  @observes("composeState")
+  _setMainOutletPaddingBottom() {
+    const mainOutletElement = document.querySelector("#main-outlet");
+    if (this.composeState === Composer.OPEN) {
+      mainOutletElement.style.paddingBottom = "300px";
```

Comment:
Maybe we use a class here instead, that way CSS ends up being less limited (eg: 300px in CSS can become 400px)

Reply:
Oh, this gave me a thought, maybe we can use a CSS custom property for `--composer-height`, and both the composer and the main outlet's padding rely on that? That can open up other possibilities as well, i.e. we can use the same property to place the floating chat element, the topic progress bar and so on (with fewer window resize listeners).